feat(javascript): Add "Data Collected" (#13527)
getsentry/sentry-docs: docs/platforms/javascript/common/data-management/data-collected/index.mdx

---
title: Data Collected
description: "See what data is collected by the Sentry SDK."
sidebar_order: 1
---

Sentry takes data privacy very seriously and has default settings in place that prioritize data safety, especially when it comes to personally identifiable information (PII) data. When you add the Sentry SDK to your application, you allow it to collect data and send it to Sentry during the runtime of your application.

The category types and amount of data collected vary, depending on the integrations you've enabled in the Sentry SDK. This page lists data categories that the Sentry Android SDK collects.

Many of the categories listed here require you to enable the <PlatformLink to="/configuration/options/#sendDefaultPii">sendDefaultPii option</PlatformLink>.

## HTTP Headers

> By default, the Sentry SDK doesn't send any HTTP response or request headers.

Reviewer: this isn't entirely true, right? IIRC from your PR, it's true for browser but in Node we always send headers? If so, let's maybe only show this for browser JS?

Author: yeah this is currently not true but I was already writing the optimistic version of the docs for after [my PR](https://github.com/getsentry/sentry-javascript/pull/16143) is merged :D

But since we don't filter sending the HTTP headers I'll rewrite this

To start sending HTTP headers, set <PlatformLink to="/configuration/options/#sendDefaultPii">`sendDefaultPii: true`</PlatformLink>.

## Cookies

By default, the Sentry SDK doesn't send cookies. Sentry tries to remove any cookies that contain sensitive information, such as the Session ID and CSRF Token cookies in Django.

If you want to send cookies, set `sendDefaultPii: true` in the `Sentry.init()` call. This will send the cookie headers `Cookie` and `Set-Cookie` from fetch and XHR requests.

## Information About Logged-in User

> By default, the Sentry SDK doesn't send any information about the logged-in user, such as email address, user ID, or username. Even if enabled, the type of logged-in user information you'll be able to send depends on the integrations you enable in Sentry's SDK. Most integrations won't send any user information. Some will only set the user ID, but there are a few (e.g. <PlatformLink to="/user-feedback/">User Feedback</PlatformLink>) that will set the user ID, username, and email address.

Reviewer: > even if enabled

if what is enabled -- `sendDefaultPii`?

> Some will only set the user ID

I'm curious myself now, which ones? 😅

Author: That's bad wording 🙈 
It basically depends on what the user provides in the object. I'm going to re-write this all 😅

> To start sending logged-in user information, set <PlatformLink to="/configuration/options/#sendDefaultPii">`sendDefaultPii: true`</PlatformLink> in your `Sentry.init({})` config.

Reviewer: Hmm maybe I'm missing something but wouldn't I have to use `setUser` to actually set user data? Other than that, there's IP inferral and stuff but this is explained in the section below

## Users' IP Address and Location

By default, the Sentry SDK doesn't send the user's IP address.

To enable sending the user's IP address and infer the location, set <PlatformLink to="/configuration/options/#sendDefaultPii">`sendDefaultPii: true`</PlatformLink>. In some integrations such as <PlatformLink to="/guides/astro/#customize-server-instrumentation">`handleRequest`</PlatformLink> in Astro, you can send the user's IP address by enabling `trackClientIp`.

If sending the IP address is enabled we will try to infer the IP address or use the IP address provided by `ip_address` in <PlatformLink to="/apis/#setUser">`Sentry.setUser()`</PlatformLink>. If you set `ip_address: null`, the IP address won't be inferred.

## Request URL

The full request URL of outgoing and incoming HTTP requests is **always sent to Sentry**. Depending on your application, this could contain PII data. For example, a URL like `/users/1234/details`, where `1234` is a user id (which may be considered PII).

## Request Query String

The full request query string of outgoing and incoming HTTP requests is **always sent to Sentry**. Depending on your application, this could contain PII data.

> 

Reviewer: two things:

l: let's add an example for a query param (and maybe also fragment?), just so that everyone understands what we're referring to

m: AFAIK we apply [server-side PII scrubbing](https://sentry-docs-git-sig-js-data-collected.sentry.dev/security-legal-pii/scrubbing/server-side-scrubbing/) to the URL query and fragment fields. This doesn't change the fact that we send the data but maybe it's worth mentioning? I guess we do this on a lot of fields mentioned on this page in general, so instead of mentioning it here specifically, we could write a short section about it and link to the docs.

> ## Request Body

Reviewer: Does this apply to all requests we collect (everywhere)?

I'd specifically mention if this applies to incoming or outgoing requests because I think there's different behaviour, right?

- **The type of the request body:**
  - JSON and HTML Form bodies are sent
- **The size of the request body:** There's a <PlatformLink to="/configuration/options/#max-request-body-size">maxRequestBodySize</PlatformLink> option that's set to `NONE` by default. This means by default no request body is sent to Sentry.

## Response Body

> By default, the Sentry SDK doesn't send the content of response bodies. By default, the SDK will send the response body size based on the `content-length` header.

Reviewer: l: I think this is also about outgoing requests, right? 

```suggestion
By default, the Sentry SDK doesn't send the body content of responses received from outgoing requests . By default, the SDK will send the response body size based on the `content-length` header.
```
does this make sense?

## Source Context

> By default, using the Sentry CLI Wizard will enable uploading source maps to Sentry.

Reviewer: l: I'd make this a bit clearer:

```suggestion
By default, SDKs set up by the Sentry CLI Wizard (`@sentry/wizard`) will enable uploading source maps to Sentry. 
```

To disable source map upload, see <PlatformLink to="/sourcemaps">the Source Maps documentation</PlatformLink>.

## Local Variables In Stack Trace

> By default, the Sentry SDK will not send local variables in the error stack trace in client-side JavaScript SDKs.

Reviewer: l: we never do this in browser as we don't get this data. 

```suggestion
The Sentry SDK does not send local variables in the error stack trace in client-side JavaScript SDKs.
```

> <PlatformSection notSupported={["javascript", "javascript.angular", "javascript.ember", "javascript.react", "javascript.solid", "javascript.svelte", "javascript.vue"]}>

Reviewer: l: I think you can use `PlatformCategorySection` to disable client-side SDKs here

## Device, Browser, OS and Runtime Information

By default, the Sentry SDK sends information about the device and runtime to Sentry.

In browser environments, this information is obtained by the User Agent string. The User Agent string contains information about the browser, operating system, and device type.

In server-side environments, the Sentry SDK uses the `os` module to get information about the operating system and architecture.

<PlatformSection notSupported={["javascript.node", "javascript.aws-lambda", "javascript.azure-functions", "javascript.connect", "javascript.express", "javascript.fastify", "javascript.gcp-functions", "javascript.hapi",  "javascript.hono", "javascript.koa", "javascript.nestjs", "javascript.wasm"]}>
  ## Referrer URL

  By default, the Sentry SDK sends the referrer URL to Sentry. This is the URL of the page that linked to the current page.

  ## Session Replay

  By default, our Session Replay SDK masks all text content, images, web views, and user input. This helps ensure that no sensitive data is exposed. You can find <PlatformLink to="/session-replay/privacy/">more details in the Session Replay documentation</PlatformLink>.

  ## Console Logs

>   By default, the Sentry SDK sends JS console logs to Sentry which may contain PII data.

Reviewer: l: maybe make this a bit more explcit? No strong opinions though :)

```suggestion
  By default, the Sentry SDK sends JS console logs to Sentry as breadcrumbs which may contain PII data.
```

---
title: Data Collected
description: "See what data is collected by the Sentry SDK."
sidebar_order: 1
notSupported:
  - javascript.electron
---

> Sentry takes data privacy very seriously and has default settings in place that prioritize data safety, especially when it comes to personally identifiable information (PII) data. When you add the Sentry SDK to your application, you allow it to collect data and send it to Sentry during the runtime of your application.

Reviewer 2: Not just runtime I would say?

Author: True, source maps is build time

> The category types and amount of data collected vary, depending on the integrations you've enabled in the Sentry SDK. This page lists data categories that the Sentry Android SDK collects.

Reviewer 2: ```suggestion
The category types and amount of data collected vary, depending on the integrations you've enabled in the Sentry SDK. This page lists data categories that the Sentry JavaScript SDK collects.
```

Many of the categories listed here require you to enable the <PlatformLink to="/configuration/options/#sendDefaultPii">sendDefaultPii option</PlatformLink>.

## HTTP Headers

By default, the Sentry SDK sends HTTP response or request headers.

## Cookies

> By default, the Sentry SDK doesn't send cookies. Sentry tries to remove any cookies that contain sensitive information, such as the Session ID and CSRF Token cookies in Django.

Reviewer 2: Should we have a more JS flavoured example than Django here maybe?

</PlatformSection>

## Device, Browser, OS and Runtime Information

By default, the Sentry SDK sends information about the device and runtime to Sentry.

In browser environments, this information is obtained by the User Agent string. The User Agent string contains information about the browser, operating system, and device type.

In server-side environments, the Sentry SDK uses the `os` module to get information about the operating system and architecture.

<PlatformSection notSupported={["javascript.azure-functions", "javascript.bun", "javascript.deno", "javascript.node", "javascript.aws-lambda", "javascript.connect", "javascript.cloudflare", "javascript.express", "javascript.fastify", "javascript.gcp-functions", "javascript.hapi",  "javascript.hono", "javascript.koa", "javascript.nestjs", "javascript.wasm"]}>
  <PlatformSection notSupported={["javascript.cordova"]}>
    ## Session Replay

    By default, our Session Replay SDK masks all text content, images, web views, and user input. This helps ensure that no sensitive data is exposed. You can find <PlatformLink to="/session-replay/privacy/">more details in the Session Replay documentation</PlatformLink>.

>     Session Replay also captures basic information about all outgoing fetch and XHR requests in your application. This includes the URL, request and response body size, method, and status code. If <PlatformLink to="/session-replay/configuration/#network-details">`networkDetailAllowUrls`</PlatformLink> are defined, the request and response body will be sent to Sentry as well. This can include PII data if the request or response body contains PII information.

Reviewer 2: Session replay also captures all console logs if that is of importance here?
